src/shared/utilities/childProcess.ts

```diff
@@ -45,138 +76,153 @@ export class ChildProcess {
+    public async run(params: ChildProcessRunOptions = {}): Promise<ChildProcessResult> {
         if (this.childProcess) {
             throw new Error('process already started')
         }
+
         this.log.info(`Running: ${this.toString()}`)
 
-        // Async.
-        // See also crossSpawn.spawnSync().
-        // Arguments are forwarded[1] to node `child_process` module, see its documentation[2].
-        // [1] https://github.com/moxystudio/node-cross-spawn/blob/master/index.js
-        // [2] https://nodejs.org/api/child_process.html
-        this.childProcess = crossSpawn.spawn(this.command, this.args, this.options)
-
-        function errorHandler(process: ChildProcess, params: ChildProcessStartArguments, error: Error): void {
-            process.processError = error
-            if (params.onError) {
-                params.onError(error)
-            }
+        const cleanup = () => {
+            this.childProcess?.removeAllListeners()
+            this.childProcess?.stdout?.removeAllListeners()
+            this.childProcess?.stderr?.removeAllListeners()
         }
 
-        // Emitted whenever:
-        //  1. Process could not be spawned, or
-        //  2. Process could not be killed, or
-        //  3. Sending a message to the child process failed.
-        // https://nodejs.org/api/child_process.html#child_process_class_childprocess
-        // We also register error event handlers on the output/error streams in case a lower level library fails
-        this.childProcess.on('error', err => {
-            errorHandler(this, params, err)
-        })
-        this.childProcess.stdout?.on('error', err => {
-            errorHandler(this, params, err)
-        })
-        this.childProcess.stderr?.on('error', err => {
-            errorHandler(this, params, err)
-        })
-
-        this.childProcess.stdout?.on('data', (data: { toString(): string }) => {
-            if (params.collect) {
-                this.stdoutChunks.push(data.toString())
-            }
+        const mergedOptions = {
+            ...this.options,
+            ...params,
+            spawnOptions: { ...this.options.spawnOptions, ...params.spawnOptions },
+        }
+        const { rejectOnError, rejectOnExit, timeout } = mergedOptions
+        const args = this.args.concat(mergedOptions.extraArgs ?? [])
 
-            if (params.onStdout) {
-                params.onStdout(data.toString())
-            }
-        })
+        // Defaults
+        mergedOptions.collect ??= true
+        mergedOptions.waitForStreams ??= true
+        mergedOptions.stopOnError ??= true
 
-        this.childProcess.stderr?.on('data', (data: { toString(): string }) => {
-            if (params.collect) {
-                this.stderrChunks.push(data.toString())
+        return new Promise<ChildProcessResult>((resolve, reject) => {
+            const errorHandler = (error: Error, force = mergedOptions.useForceStop) => {
```

Comment:
Assuming `force` isn't settable elsewhere, so no point in including as a param and setting the default value. Just use the default directly.

Reply:
It's used in the case of a timeout, though I'm thinking we should force kill on an error as well.